### DISCUSSION TITLE
Who uses SpotBugs?

### DISCUSSION THREAD

**Discussion by KengoTODA** [spotbugs/spotbugs#1607] [General]
It's just my personal curiosity inspired by cucumber/common/issues/568, but please tell me your usage if you're using SpotBugs for your project/organization.

I don't know actual usage so much, what I know is just the [Maven Central stats](https://github.com/spotbugs/discuss/issues/5), [public repos depending on the Maven plugin](https://github.com/spotbugs/spotbugs-gradle-plugin/network/dependents?dependent_type=REPOSITORY) and [Eclipse Plugin Marketplace stats](https://marketplace.eclipse.org/content/spotbugs-eclipse-plugin#group-metrics-tab). All of them don't tell detailed usage and how they contribute to your daily hacking.

Your usage share would motivate me (and probably others) to support the community.
Thanks in advance! 😃

**Comment by iloveeclipse** [comment]
We at [Advantest](https://www.eclipse.org/membership/showMember.php?member_id=1329) use SpotBugs ant task for continuous integration and Eclipse plugin, since ~10 years or so. Means, ~50+ builds per day & on every file save in editor.

**Comment by KengoTODA** [comment]
I'll list what I know. Several SAST solutions use SpotBugs as follows:

* [Sonatype Lift](https://help.sonatype.com/lift/configuring-lift#ConfiguringLift-Java:SpecifyingTools) uses [find-sec-bugs](https://github.com/find-sec-bugs/find-sec-bugs)
* [GitLab](https://gitlab.com/gitlab-org/security-products/analyzers/spotbugs) uses SpotBugs and [find-sec-bugs](https://github.com/find-sec-bugs/find-sec-bugs)
* [Codacy](https://docs.codacy.com/related-tools/local-analysis/running-spotbugs/)
* [Mixeway](https://mixeway.io/2020/08/28/mixeway-scanner-aggregator/)

**Comment by ThrawnCA** [comment]
My team uses it to check various Queensland Government services, such as [Right To Information applications](https://www.smartservice.qld.gov.au/services/information-requests/home).

**Comment by h3xstream** [comment]
I know that Find Security Bugs (with SpotBugs) is used in those organizations:

- Companies/Origanisation: Netflix, Atlassian, Liferay, Gitlab and OpenDaylight
- US Gov: SWAMP Project (Department of Defense), US National Institutes of Health and Department of the Navy

--- 
**Fun Fact**

 SpotBugs (FindBugs) was used in the security audit of Norvegian's internet voting system.

> An early step taken to get better acquainted with the code base, was by running automated
> tools for static analysis. In particular, FindBugs7 with the “Find Security Bugs”-plugin, as
> well as PMD8, were used. These are freely available tools that are simple to set up and run,
> which use heuristic techniques to discover possible or probable code errors. While this type
> of general source code review in general is beside the scope of this audit, it has been a useful
> technique to discover possible problem areas and catch low-hanging fruit. In particular,
> FindBugs has been quite helpful.

Source: https://www.regjeringen.no/globalassets/upload/krd/prosjekter/e-valg/kildekode/evalg_rapport_kildekodegjennomgang.pdf

--- 

I remember FindBugs being quoted directly in the OWASP Top 2013 (expired in 2017). It didn't make sense to mention a specific tool in this high-level document. Nevertheless, it was great mention.
SpotBugs and FSB are still quoted in some OWASP guides.

**Comment by tomas-langer** [comment]
We use Spotbugs in Helidon (https://github.com/oracle/helidon). We also use the " Find Security Bugs" plugin.

**Comment by dkrupp** [comment]
We use Spotbugs and FindsecBugs in Ericsson. Thanks for your work!
